internal/common/function/function.go

```diff
@@ -77,66 +77,125 @@ func ValidateBackupSpec(nonAdminBackup *nacv1alpha1.NonAdminBackup) error {
+	// Validate input parameters
+	if namespace == constant.EmptyString || labelKey == constant.EmptyString || labelValue == constant.EmptyString {
+		return fmt.Errorf("invalid input: namespace, labelKey, and labelValue must not be empty")
 	}
 
-	return fmt.Sprintf("%s-%s-%s", constant.VeleroBackupNamePrefix, namespace, nameHash)
+	labelSelector := labels.SelectorFromSet(labels.Set{labelKey: labelValue})
+
+	// Attempt to list objects with the specified label
+	if err := clientInstance.List(ctx, objectList, &client.ListOptions{
+		LabelSelector: labelSelector,
+		Namespace:     namespace,
+	}); err != nil {
+		return fmt.Errorf("failed to list objects in namespace '%s': %w", namespace, err)
+	}
+
+	return nil
+}
+
+// GetVeleroBackupByLabel retrieves a VeleroBackup object based on a specified label within a given namespace.
+// It returns the VeleroBackup only when exactly one object is found, throws an error if multiple backups are found,
+// or returns nil if no matches are found.
+func GetVeleroBackupByLabel(ctx context.Context, clientInstance client.Client, namespace string, labelValue string) (*velerov1.Backup, error) {
+	veleroBackupList := &velerov1.BackupList{}
+
+	// Call the generic GetObjectByLabel function
+	if err := GetObjectByLabel(ctx, clientInstance, namespace, constant.NabOriginUUIDLabel, labelValue, veleroBackupList); err != nil {
+		return nil, err
+	}
+
+	switch len(veleroBackupList.Items) {
+	case 0:
+		return nil, nil // No matching VeleroBackup found
+	case 1:
+		return &veleroBackupList.Items[0], nil // Found 1 matching VeleroBackup
+	default:
+		return nil, fmt.Errorf("multiple VeleroBackup objects found with label %s=%s in namespace '%s'", constant.NabOriginUUIDLabel, labelValue, namespace)
+	}
 }
 
 // CheckVeleroBackupMetadata return true if Velero Backup object has required Non Admin labels and annotations, false otherwise
 func CheckVeleroBackupMetadata(obj client.Object) bool {
-	labels := obj.GetLabels()
-	if !checkLabelValue(labels, constant.OadpLabel, constant.OadpLabelValue) {
+	objLabels := obj.GetLabels()
+	if !checkLabelValue(objLabels, constant.OadpLabel, constant.OadpLabelValue) {
 		return false
 	}
-	if !checkLabelValue(labels, constant.ManagedByLabel, constant.ManagedByLabelValue) {
+	if !checkLabelValue(objLabels, constant.ManagedByLabel, constant.ManagedByLabelValue) {
 		return false
 	}
 
-	annotations := obj.GetAnnotations()
-	if !checkAnnotationValueIsValid(annotations, constant.NabOriginNamespaceAnnotation) {
+	if !checkAnnotationOrLabelsValueIsValid(objLabels, constant.NabOriginUUIDLabel) {
 		return false
 	}
-	if !checkAnnotationValueIsValid(annotations, constant.NabOriginNameAnnotation) {
+
+	annotations := obj.GetAnnotations()
+	if !checkAnnotationOrLabelsValueIsValid(annotations, constant.NabOriginNamespaceAnnotation) {
 		return false
 	}
-	// TODO what is a valid uuid?
```

Comment:
I think this is a bad change

First, the function does check for max label length

It also does not check if UUID value is valid

I would not change `checkAnnotationValueIsValid` function logic

Correct approach here is to create new function to check if label value is valid and also check if UUID value is valid, by using Parse function https://pkg.go.dev/github.com/google/uuid@v1.6.0#Parse, right?

Reply:
The Non admin UUID is not part of annotation, it's part of label now. Also it's not a valid UUID as we add more to it (prefix).

Reply:
RE: "First, the function does check for max label length"

It does by comparing it in `if len(nacObjectName) > constant.MaximumNacObjectNameLength {`

Reply:
253 char limitation for annotation is needed here because the annotations values are kubernetes names, so if it longer than that, we can not create object with that value (sync controller for example)

Reply:
Can you point me to the code where currently annotation value is used as the base for *creating*  k8s object with the name as the value ? I don't think there is, we are using it to store already existing name but not to create.

Reply:
I agree with that (please link issue #90 to this PR), but this discussion is another problem (validation of annotation inside velero backup predicate)

Reply:
updated the PR description to link #90